org.osgi.service.jakartaws/src/org/osgi/service/jakartaws/runtime/dto/HandlerDTO.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0 
+ *******************************************************************************/
+package org.osgi.service.jakartaws.runtime.dto;
+
+import org.osgi.dto.DTO;
+import org.osgi.framework.dto.ServiceReferenceDTO;
+
+/**
+ * Represents a handler currently known to the webservice runtime
+ */
+public class HandlerDTO extends DTO {
+	/**
+	 * The service reference of the handler, is never {@code null}.
+	 */
+	public ServiceReferenceDTO serviceReference;
```

Comment:
maybe just use the  `serviceId` like in filter

https://github.com/osgi/osgi/blob/b32032ea3bb23de7d3b27622dc3865dfc803cef5/org.osgi.service.servlet/src/org/osgi/service/servlet/runtime/dto/FilterDTO.java#L87C1-L96C1

Reply:
A service reference can easily transformed into a DTO and offer much more information (e.g. properties) as the requested error reporting capabilities will as well require to fetch service references it also seem to restrict the implementation to have the full information available here.